plugin/sqlsh/src/dist/sqlsh.js

```diff
@@ -44,54 +44,149 @@ define(function(require, exports, module) {
             ecb("failed to create shell");
           });
     },
-    evaluate: function(code, modelOutput,refreshObj) {
-      var deferred = Q.defer();
-      
-      if (SqlShCancelFunction) {
-        deferred.reject("An evaluation is already in progress");
-        return deferred.promise;
-      }
+    
+    evaluateWithPassword : function(code, modelOutput, refreshObj, deferred) {
+    	if (SqlShCancelFunction) {
+    		deferred.reject("An evaluation is already in progress");
+    		return deferred.promise;
+    	}
+    	var self = this;
+    	bkHelper.setupProgressOutput(modelOutput);
+        
+    	$.ajax({
+    		type: "POST",
+    		datatype: "json",
+    		url: bkHelper.serverUrl(serviceBase + "/rest/sqlsh/evaluate"),
+    		data: {shellId: self.settings.shellID, code: code}
+    	}).done(function(ret) {
+    		SqlShCancelFunction = function () {
+    			$.ajax({
+    				type: "POST",
+    				datatype: "json",
+    				url: bkHelper.serverUrl(serviceBase + "/rest/sqlsh/cancelExecution"),
+    				data: {shellId: self.settings.shellID}
+    			}).done(function (ret) {
+    				console.log("done cancelExecution",ret);
+    			});
+    			bkHelper.setupCancellingOutput(modelOutput);
+    		}
+    		var onEvalStatusUpdate = function(evaluation) {
+    			if (bkHelper.receiveEvaluationUpdate(modelOutput, evaluation, PLUGIN_NAME, self.settings.shellID)) {
+    				cometdUtil.unsubscribe(evaluation.update_id);
+    				SqlShCancelFunction = null;
+    				if (evaluation.status === "ERROR")
+    					deferred.reject(evaluation.payload);
+    				else
+    					deferred.resolve(evaluation.payload);
+    			}
+    			if (refreshObj !== undefined)
+    				refreshObj.outputRefreshed();
+    			else
+    				bkHelper.refreshRootScope();
+    		};
+    		onEvalStatusUpdate(ret);
+    		if (ret.update_id) {
+    			cometdUtil.subscribe(ret.update_id, onEvalStatusUpdate);
+    		}
+    	});
+    },
+    
+    evaluate: function(code, modelOutput, refreshObj) {
+    	var self = this;
+    	var deferred = Q.defer();
+    	return getListOfConnectiononWhoNeedDialog(self.settings.shellID).success(function(ret) {
+    		
+    		if(ret.length > 0){
+    			for (var i=0; i<ret.length; i++) {
+    				
+    				var html = '';
+    				if(ret[i].connectionName == null){
+    					html += '<input id="connection_name" type="hidden"/>';
+    				}else{
+    					html += '<input id="connection_name" type="hidden" value="' + ret[i].connectionName + '" />';
+    				}
+    				html += '<table class="table">';
```

Comment:
can this template be moved to the helper.js and defined with a regular jst.html?